docs/topics/api/addons.rst

```diff
@@ -0,0 +1,42 @@
+=======
+Add-ons
+=======
+
+.. note::
+
+    These APIs are experimental and are currently being worked on. Endpoints
+    may change without warning.
+
+------
+Search
+------
+
+.. _addon-search:
+
+This endpoint allows you to search through public add-ons.
+
+.. http:get:: /api/v3/addons/search/
+
+    :param q: The search query. **Not implemented yet**.
+    :>json count: The number of results for this query.
+    :>json next: The URL of the next page of results.
+    :>json previous: The URL of the previous page of results.
+    :>json results: An array of :ref:`add-ons <addon-detail>`.
+
+------
+Detail
+------
+
+.. _addon-detail:
+
+This endpoint allows you to fetch a specific add-on by id, slug or guid.
+
+**Not implemented yet**.
+
+.. http:get:: /api/v3/addons/addon/(int:id|string:slug|string:guid)/
```

Comment:
Are there any slugs and guids that match? I'm guessing the `{}@` requirement in the guid makes it an invalid slug but just wondering.